dora/src/semck/stdlib.rs

```diff
@@ -239,6 +263,31 @@ fn internal_struct(
     struct_id
 }
 
+fn internal_annotation(
+    vm: &mut VM,
+    namespace_id: NamespaceId,
+    name: &str,
+    internal_annotation: Modifier,
+) -> AnnotationId {
+    let name = vm.interner.intern(name);
+    let mut annotations = vm.annotations.lock();
+    let id: AnnotationId = annotations.len().into();
+    let annotation = Annotation {
+        id,
+        name,
+        file_id: FileId::from(0),
+        pos: Position::invalid(),
```

Comment:
I know I've used `Position::new(1, 1)` a lot, but `Position::invalid()` violates the asserts in the `Position::new` constructor. If we really need something like invalid positions, imho we should leverage the type system and use `Option<Position>`.

Reply:
I'm fine with `Position::new(1, 1)`, the idea with an illegal position was that it was immediately obvious that this things was not correctly initialized. Making it an `Option` burdens the callers with a possibility they should never have to deal with (because the invalid state is transient, the next phase is supposed to fill in the correct details).